compiler/il/OMRNode.cpp

```diff
@@ -3083,28 +3087,77 @@ OMR::Node::getMonitorClass(TR_ResolvedMethod * vmMethod)
    return 0;
    }
 
+void *
+OMR::Node::getMonitorInfoInNode()
+   {
+   TR_ASSERT(self()->hasMonitorInfoInNode(),"Node %p does not have monitor info in Node");
```

Comment:
Even if the below is not ASSERT_FATAL surely this need to be? Otherwise you are going to read garbage or potentially read an unallocated child right?